Travis CI failed since #13889

## CI

Refer to:
#13889
https://app.travis-ci.com/github/apache/shardingsphere/builds/242957214

```
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ shardingsphere-proxy-backend ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.shardingsphere.proxy.backend.response.data.QueryResponseRowTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.984 s - in org.apache.shardingsphere.proxy.backend.response.data.QueryResponseRowTest
[INFO] Running org.apache.shardingsphere.proxy.backend.response.header.query.impl.QueryHeaderBuilderTest
[INFO] Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.596 s - in org.apache.shardingsphere.proxy.backend.response.header.query.impl.QueryHeaderBuilderTest
[INFO] Running org.apache.shardingsphere.proxy.backend.response.header.update.UpdateResponseHeaderTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.061 s - in org.apache.shardingsphere.proxy.backend.response.header.update.UpdateResponseHeaderTest
[INFO] Running org.apache.shardingsphere.proxy.backend.exception.TableModifyInTransactionExceptionTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.281 s - in org.apache.shardingsphere.proxy.backend.exception.TableModifyInTransactionExceptionTest
[INFO] Running org.apache.shardingsphere.proxy.backend.context.BackendExecutorContextTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.968 s - in org.apache.shardingsphere.proxy.backend.context.BackendExecutorContextTest
[INFO] Running org.apache.shardingsphere.proxy.backend.context.ProxyContextTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.083 s - in org.apache.shardingsphere.proxy.backend.context.ProxyContextTest
[INFO] Running org.apache.shardingsphere.proxy.backend.util.SystemPropertyUtilTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in org.apache.shardingsphere.proxy.backend.util.SystemPropertyUtilTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.decorator.HikariJDBCParameterDecoratorTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.335 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.decorator.HikariJDBCParameterDecoratorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.JDBCBackendDataSourceTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.445 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.JDBCBackendDataSourceTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.statement.impl.MySQLStatementMemoryStrictlyFetchSizeSetterTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.329 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.statement.impl.MySQLStatementMemoryStrictlyFetchSizeSetterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.statement.impl.PostgreSQLStatementMemoryStrictlyFetchSizeSetterTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.009 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.statement.impl.PostgreSQLStatementMemoryStrictlyFetchSizeSetterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.statement.impl.OpenGaussStatementMemoryStrictlyFetchSizeSetterTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.statement.impl.OpenGaussStatementMemoryStrictlyFetchSizeSetterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.SQLServerRecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.SQLServerRecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.P6SpyDriverRecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.P6SpyDriverRecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.PostgreSQLRecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.PostgreSQLRecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.H2RecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.H2RecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.OpenGaussRecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.OpenGaussRecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.MySQLRecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.MySQLRecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.OracleRecognizerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.impl.OracleRecognizerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.JDBCDriverURLRecognizerEngineTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.recognizer.JDBCDriverURLRecognizerEngineTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.LocalTransactionManagerTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.218 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.LocalTransactionManagerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.TransactionStatusTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.TransactionStatusTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.BackendTransactionManagerTest
[INFO] Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.213 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.BackendTransactionManagerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.ConnectionStatusTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.22 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.ConnectionStatusTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.ConnectionUrlParserTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.ConnectionUrlParserTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.ResourceLockTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.252 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.ResourceLockTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.BackendConnectionTest
[INFO] Tests run: 22, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.712 s - in org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.BackendConnectionTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutorTest
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.284 s - in org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngineFactoryTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.134 s - in org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngineFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngineTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.481 s - in org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngineTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest
[ERROR] Tests run: 15, Failures: 0, Errors: 14, Skipped: 0, Time elapsed: 0.225 s <<< FAILURE! - in org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest
[ERROR] assertNewInstanceWithSetAutoCommitToOff(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.136 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSetAutoCommitToOff(TextProtocolBackendHandlerFactoryTest.java:138)
[ERROR] assertNewInstanceWithShowDatabase(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.006 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithShowDatabase(TextProtocolBackendHandlerFactoryTest.java:173)
[ERROR] assertNewInstanceWithErrorRDL(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.005 s  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.shardingsphere.sql.parser.exception.SQLParsingException> but was<java.lang.NullPointerException>
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithErrorRDL(TextProtocolBackendHandlerFactoryTest.java:227)
[ERROR] assertNewInstanceWithErrorSQL(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.004 s  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.shardingsphere.sql.parser.exception.SQLParsingException> but was<java.lang.NullPointerException>
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithErrorSQL(TextProtocolBackendHandlerFactoryTest.java:220)
[ERROR] assertNewInstanceWithScopeSetAutoCommitToOnForInTransaction(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.006 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithScopeSetAutoCommitToOnForInTransaction(TextProtocolBackendHandlerFactoryTest.java:159)
[ERROR] assertNewInstanceWithSet(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.004 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSet(TextProtocolBackendHandlerFactoryTest.java:180)
[ERROR] assertNewInstanceWithUse(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.005 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithUse(TextProtocolBackendHandlerFactoryTest.java:166)
[ERROR] assertNewInstanceWithStartTransaction(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.005 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithStartTransaction(TextProtocolBackendHandlerFactoryTest.java:131)
[ERROR] assertNewInstanceWithCommonDistSQL(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithCommonDistSQL(TextProtocolBackendHandlerFactoryTest.java:108)
[ERROR] assertNewInstanceWithSetAutoCommitToOn(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.004 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSetAutoCommitToOn(TextProtocolBackendHandlerFactoryTest.java:152)
[ERROR] assertNewInstanceWithShow(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithShow(TextProtocolBackendHandlerFactoryTest.java:187)
[ERROR] assertNewInstanceWithBegin(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithBegin(TextProtocolBackendHandlerFactoryTest.java:124)
[ERROR] assertNewInstanceWithQuery(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithQuery(TextProtocolBackendHandlerFactoryTest.java:203)
[ERROR] assertNewInstanceWithScopeSetAutoCommitToOff(org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithScopeSetAutoCommitToOff(TextProtocolBackendHandlerFactoryTest.java:145)
[INFO] Running org.apache.shardingsphere.proxy.backend.text.data.impl.BroadcastDatabaseBackendHandlerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.061 s - in org.apache.shardingsphere.proxy.backend.text.data.impl.BroadcastDatabaseBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.data.impl.UnicastDatabaseBackendHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.019 s - in org.apache.shardingsphere.proxy.backend.text.data.impl.UnicastDatabaseBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.011 s - in org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.data.DatabaseBackendHandlerFactoryTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.294 s - in org.apache.shardingsphere.proxy.backend.text.data.DatabaseBackendHandlerFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.advance.ParseDistSQLBackendHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.636 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.advance.ParseDistSQLBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.ShowVariableBackendHandlerTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.014 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.ShowVariableBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.SetVariableBackendHandlerTest
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.029 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.SetVariableBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.set.executor.SetVariableExecutorTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.set.executor.SetVariableExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.set.SetStatementExecutorFactoryTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.285 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.set.SetStatementExecutorFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.hint.HintStatementExecutorFactoryTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.499 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.hint.HintStatementExecutorFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.ral.ShowAllVariablesBackendHandlerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.009 s - in org.apache.shardingsphere.proxy.backend.text.distsql.ral.ShowAllVariablesBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.DistSQLBackendHandlerFactoryTest
[INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.395 s - in org.apache.shardingsphere.proxy.backend.text.distsql.DistSQLBackendHandlerFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rql.SingleTableRulesQueryResultSetTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.057 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rql.SingleTableRulesQueryResultSetTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rql.RuleDefinitionBackendHandlerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rql.RuleDefinitionBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rql.DataSourceQueryResultSetTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.235 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rql.DataSourceQueryResultSetTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rql.SingleTableQueryResultSetTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.137 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rql.SingleTableQueryResultSetTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rql.RQLBackendHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.11 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rql.RQLBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rql.DataSourceQueryResultSetConverterTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rql.DataSourceQueryResultSetConverterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AlterResourceBackendHandlerTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.09 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AlterResourceBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.DropResourceBackendHandlerTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.178 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.DropResourceBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.ResourceSegmentsConverterTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.ResourceSegmentsConverterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.CreateDefaultSingleTableRuleUpdaterTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.088 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.CreateDefaultSingleTableRuleUpdaterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.AlterDefaultSingleTableRuleUpdaterTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.AlterDefaultSingleTableRuleUpdaterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.DropDefaultSingleTableRuleUpdaterTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.DropDefaultSingleTableRuleUpdaterTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.postgresql.PostgreSQLAdminExecutorFactoryTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.apache.shardingsphere.proxy.backend.text.admin.postgresql.PostgreSQLAdminExecutorFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.postgresql.executor.SelectDatabaseExecutorTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.262 s - in org.apache.shardingsphere.proxy.backend.text.admin.postgresql.executor.SelectDatabaseExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.postgresql.executor.SelectTableExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.015 s - in org.apache.shardingsphere.proxy.backend.text.admin.postgresql.executor.SelectTableExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.DatabaseOperateBackendHandlerFactoryTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.017 s - in org.apache.shardingsphere.proxy.backend.text.admin.DatabaseOperateBackendHandlerFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.MySQLAdminExecutorFactoryTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.289 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.MySQLAdminExecutorFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowCurrentUserExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.175 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowCurrentUserExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowProcessListExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.199 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowProcessListExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.UseDatabaseExecutorTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.131 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.UseDatabaseExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowConnectionIdExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowConnectionIdExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowFunctionStatusExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowFunctionStatusExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowTablesExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowTablesExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowVersionExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.202 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowVersionExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowTablesStatusExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowTablesStatusExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.information.SelectInformationExecutorTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.073 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.information.SelectInformationExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowDatabasesExecutorTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.038 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowDatabasesExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowCreateDatabaseExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowCreateDatabaseExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowProcedureStatusExecutorTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in org.apache.shardingsphere.proxy.backend.text.admin.mysql.executor.ShowProcedureStatusExecutorTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.transaction.TransactionBackendHandlerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.089 s - in org.apache.shardingsphere.proxy.backend.text.transaction.TransactionBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.transaction.TransactionBackendHandlerFactoryTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.15 s - in org.apache.shardingsphere.proxy.backend.text.transaction.TransactionBackendHandlerFactoryTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.database.DropDatabaseBackendHandlerTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.135 s - in org.apache.shardingsphere.proxy.backend.text.database.DropDatabaseBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.database.CreateDatabaseBackendHandlerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.085 s - in org.apache.shardingsphere.proxy.backend.text.database.CreateDatabaseBackendHandlerTest
[INFO] Running org.apache.shardingsphere.proxy.backend.text.skip.SkipBackendHandlerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 s - in org.apache.shardingsphere.proxy.backend.text.skip.SkipBackendHandlerTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithBegin:124 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithCommonDistSQL:108 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithErrorRDL »  Unexpec...
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithErrorSQL »  Unexpec...
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithQuery:203 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithScopeSetAutoCommitToOff:145 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithScopeSetAutoCommitToOnForInTransaction:159 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSet:180 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSetAutoCommitToOff:138 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSetAutoCommitToOn:152 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithShow:187 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithShowDatabase:173 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithStartTransaction:131 » NullPointer
[ERROR]   TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithUse:166 » NullPointer
[INFO] 
[ERROR] Tests run: 291, Failures: 0, Errors: 14, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for shardingsphere 5.0.1-SNAPSHOT:
[INFO] 
[INFO] shardingsphere ..................................... SUCCESS [  6.328 s]
[INFO] shardingsphere-spi ................................. SUCCESS [ 10.379 s]
[INFO] shardingsphere-sql-parser .......................... SUCCESS [  0.453 s]
[INFO] shardingsphere-sql-parser-spi ...................... SUCCESS [  2.772 s]
[INFO] shardingsphere-sql-parser-statement ................ SUCCESS [ 18.873 s]
[INFO] shardingsphere-sql-parser-engine ................... SUCCESS [ 13.171 s]
[INFO] shardingsphere-distsql ............................. SUCCESS [  0.334 s]
[INFO] shardingsphere-distsql-statement ................... SUCCESS [  2.319 s]
[INFO] shardingsphere-distsql-parser ...................... SUCCESS [  4.334 s]
[INFO] shardingsphere-features ............................ SUCCESS [  0.388 s]
[INFO] shardingsphere-sharding ............................ SUCCESS [  0.365 s]
[INFO] shardingsphere-sharding-distsql .................... SUCCESS [  0.376 s]
[INFO] shardingsphere-sharding-distsql-statement .......... SUCCESS [  2.360 s]
[INFO] shardingsphere-sharding-distsql-parser ............. SUCCESS [  3.351 s]
[INFO] shardingsphere-readwrite-splitting ................. SUCCESS [  0.395 s]
[INFO] shardingsphere-readwrite-splitting-distsql ......... SUCCESS [  0.353 s]
[INFO] shardingsphere-readwrite-splitting-distsql-statement SUCCESS [  2.260 s]
[INFO] shardingsphere-readwrite-splitting-distsql-parser .. SUCCESS [  2.868 s]
[INFO] shardingsphere-db-discovery ........................ SUCCESS [  0.340 s]
[INFO] shardingsphere-db-discovery-distsql ................ SUCCESS [  0.348 s]
[INFO] shardingsphere-db-discovery-distsql-statement ...... SUCCESS [  1.878 s]
[INFO] shardingsphere-db-discovery-distsql-parser ......... SUCCESS [  2.347 s]
[INFO] shardingsphere-encrypt ............................. SUCCESS [  0.322 s]
[INFO] shardingsphere-encrypt-distsql ..................... SUCCESS [  0.282 s]
[INFO] shardingsphere-encrypt-distsql-statement ........... SUCCESS [  2.255 s]
[INFO] shardingsphere-encrypt-distsql-parser .............. SUCCESS [  2.545 s]
[INFO] shardingsphere-scaling ............................. SUCCESS [  0.279 s]
[INFO] shardingsphere-scaling-distsql ..................... SUCCESS [  0.304 s]
[INFO] shardingsphere-scaling-distsql-statement ........... SUCCESS [  1.996 s]
[INFO] shardingsphere-scaling-distsql-parser .............. SUCCESS [  2.441 s]
[INFO] shardingsphere-shadow .............................. SUCCESS [  0.267 s]
[INFO] shardingsphere-shadow-distsql ...................... SUCCESS [  0.295 s]
[INFO] shardingsphere-shadow-distsql-statement ............ SUCCESS [  1.941 s]
[INFO] shardingsphere-shadow-distsql-parser ............... SUCCESS [  3.035 s]
[INFO] shardingsphere-test ................................ SUCCESS [  0.323 s]
[INFO] shardingsphere-parser-test ......................... SUCCESS [ 25.667 s]
[INFO] shardingsphere-sql-parser-dialect .................. SUCCESS [  0.271 s]
[INFO] shardingsphere-sql-parser-sql92 .................... SUCCESS [ 25.448 s]
[INFO] shardingsphere-sql-parser-mysql .................... SUCCESS [ 49.498 s]
[INFO] shardingsphere-sql-parser-postgresql ............... SUCCESS [ 41.888 s]
[INFO] shardingsphere-sql-parser-opengauss ................ SUCCESS [ 41.261 s]
[INFO] shardingsphere-sql-parser-oracle ................... SUCCESS [ 46.195 s]
[INFO] shardingsphere-sql-parser-sqlserver ................ SUCCESS [ 31.095 s]
[INFO] shardingsphere-db-protocol ......................... SUCCESS [  0.264 s]
[INFO] shardingsphere-db-protocol-core .................... SUCCESS [ 13.186 s]
[INFO] shardingsphere-db-protocol-mysql ................... SUCCESS [ 21.875 s]
[INFO] shardingsphere-db-protocol-postgresql .............. SUCCESS [ 17.328 s]
[INFO] shardingsphere-db-protocol-opengauss ............... SUCCESS [ 11.301 s]
[INFO] shardingsphere-test-common ......................... SUCCESS [  0.751 s]
[INFO] shardingsphere-infra ............................... SUCCESS [  0.238 s]
[INFO] shardingsphere-infra-common ........................ SUCCESS [ 27.127 s]
[INFO] shardingsphere-kernel .............................. SUCCESS [  0.271 s]
[INFO] shardingsphere-parser .............................. SUCCESS [  0.246 s]
[INFO] shardingsphere-parser-api .......................... SUCCESS [  1.676 s]
[INFO] shardingsphere-parser-core ......................... SUCCESS [  6.088 s]
[INFO] shardingsphere-infra-parser ........................ SUCCESS [  9.793 s]
[INFO] shardingsphere-infra-binder ........................ SUCCESS [ 24.527 s]
[INFO] shardingsphere-infra-route ......................... SUCCESS [ 12.719 s]
[INFO] shardingsphere-infra-rewrite ....................... SUCCESS [ 13.925 s]
[INFO] shardingsphere-infra-executor ...................... SUCCESS [ 19.265 s]
[INFO] shardingsphere-infra-merge ......................... SUCCESS [ 13.440 s]
[INFO] shardingsphere-authority ........................... SUCCESS [  0.252 s]
[INFO] shardingsphere-authority-api ....................... SUCCESS [  2.082 s]
[INFO] shardingsphere-authority-core ...................... SUCCESS [ 17.653 s]
[INFO] shardingsphere-transaction ......................... SUCCESS [  0.204 s]
[INFO] shardingsphere-transaction-api ..................... SUCCESS [  1.889 s]
[INFO] shardingsphere-transaction-core .................... SUCCESS [ 11.735 s]
[INFO] shardingsphere-infra-federation .................... SUCCESS [  0.272 s]
[INFO] shardingsphere-infra-federation-optimizer .......... SUCCESS [ 34.968 s]
[INFO] shardingsphere-infra-context ....................... SUCCESS [ 14.256 s]
[INFO] shardingsphere-infra-datetime ...................... SUCCESS [  0.252 s]
[INFO] shardingsphere-infra-datetime-spi .................. SUCCESS [  2.281 s]
[INFO] shardingsphere-infra-datetime-type ................. SUCCESS [  0.251 s]
[INFO] shardingsphere-system-datetime ..................... SUCCESS [  5.859 s]
[INFO] shardingsphere-sharding-api ........................ SUCCESS [  2.602 s]
[INFO] shardingsphere-sharding-distsql-handler ............ SUCCESS [ 17.942 s]
[INFO] shardingsphere-single-table ........................ SUCCESS [  0.235 s]
[INFO] shardingsphere-single-table-api .................... SUCCESS [  1.963 s]
[INFO] shardingsphere-single-table-core ................... SUCCESS [ 16.085 s]
[INFO] shardingsphere-sharding-core ....................... SUCCESS [01:04 min]
[INFO] shardingsphere-database-datetime ................... SUCCESS [ 13.075 s]
[INFO] shardingsphere-infra-federation-executor ........... SUCCESS [  2.804 s]
[INFO] shardingsphere-mode ................................ SUCCESS [  0.248 s]
[INFO] shardingsphere-mode-core ........................... SUCCESS [ 22.194 s]
[INFO] shardingsphere-mode-type ........................... SUCCESS [  0.226 s]
[INFO] shardingsphere-memory-mode ......................... SUCCESS [  0.213 s]
[INFO] shardingsphere-memory-mode-core .................... SUCCESS [  2.442 s]
[INFO] shardingsphere-standalone-mode ..................... SUCCESS [  0.275 s]
[INFO] shardingsphere-standalone-mode-repository .......... SUCCESS [  0.269 s]
[INFO] shardingsphere-standalone-mode-repository-api ...... SUCCESS [  1.945 s]
[INFO] shardingsphere-standalone-mode-core ................ SUCCESS [ 21.325 s]
[INFO] shardingsphere-standalone-mode-repository-provider . SUCCESS [  0.263 s]
[INFO] shardingsphere-standalone-mode-repository-file ..... SUCCESS [  6.534 s]
[INFO] shardingsphere-cluster-mode ........................ SUCCESS [  0.205 s]
[INFO] shardingsphere-cluster-mode-repository ............. SUCCESS [  0.237 s]
[INFO] shardingsphere-cluster-mode-repository-api ......... SUCCESS [  2.332 s]
[INFO] shardingsphere-cluster-mode-core ................... SUCCESS [ 26.500 s]
[INFO] shardingsphere-cluster-mode-repository-provider .... SUCCESS [  0.244 s]
[INFO] shardingsphere-cluster-mode-repository-zookeeper-curator SUCCESS [ 16.685 s]
[INFO] shardingsphere-cluster-mode-repository-etcd ........ SUCCESS [ 22.526 s]
[INFO] shardingsphere-transaction-type .................... SUCCESS [  0.246 s]
[INFO] shardingsphere-transaction-xa ...................... SUCCESS [  0.273 s]
[INFO] shardingsphere-transaction-xa-spi .................. SUCCESS [ 10.122 s]
[INFO] shardingsphere-transaction-xa-provider ............. SUCCESS [  0.179 s]
[INFO] shardingsphere-transaction-xa-atomikos ............. SUCCESS [ 11.131 s]
[INFO] shardingsphere-transaction-xa-core ................. SUCCESS [ 19.360 s]
[INFO] shardingsphere-transaction-xa-narayana ............. SUCCESS [ 12.129 s]
[INFO] shardingsphere-transaction-xa-bitronix ............. SUCCESS [ 11.526 s]
[INFO] shardingsphere-transaction-base .................... SUCCESS [  0.178 s]
[INFO] shardingsphere-transaction-base-seata-at ........... SUCCESS [ 15.311 s]
[INFO] shardingsphere-schedule ............................ SUCCESS [  0.177 s]
[INFO] shardingsphere-schedule-core ....................... SUCCESS [  1.755 s]
[INFO] shardingsphere-cdc ................................. SUCCESS [  0.207 s]
[INFO] shardingsphere-cdc-core ............................ SUCCESS [ 18.609 s]
[INFO] shardingsphere-cdc-dialect ......................... SUCCESS [  0.173 s]
[INFO] shardingsphere-cdc-mysql ........................... SUCCESS [ 20.323 s]
[INFO] shardingsphere-cdc-postgresql ...................... SUCCESS [ 18.439 s]
[INFO] shardingsphere-cdc-opengauss ....................... SUCCESS [ 11.759 s]
[INFO] shardingsphere-migration-common .................... SUCCESS [  2.057 s]
[INFO] shardingsphere-readwrite-splitting-api ............. SUCCESS [  1.790 s]
[INFO] shardingsphere-readwrite-splitting-distsql-handler . SUCCESS [ 11.517 s]
[INFO] shardingsphere-readwrite-splitting-core ............ SUCCESS [ 14.097 s]
[INFO] shardingsphere-db-discovery-api .................... SUCCESS [  1.696 s]
[INFO] shardingsphere-db-discovery-distsql-handler ........ SUCCESS [ 11.676 s]
[INFO] shardingsphere-db-discovery-provider ............... SUCCESS [  0.177 s]
[INFO] shardingsphere-db-discovery-mgr .................... SUCCESS [ 15.529 s]
[INFO] shardingsphere-db-discovery-opengauss .............. SUCCESS [ 15.495 s]
[INFO] shardingsphere-db-discovery-core ................... SUCCESS [ 14.456 s]
[INFO] shardingsphere-encrypt-api ......................... SUCCESS [  1.803 s]
[INFO] shardingsphere-encrypt-distsql-handler ............. SUCCESS [ 11.426 s]
[INFO] shardingsphere-encrypt-core ........................ SUCCESS [ 22.656 s]
[INFO] shardingsphere-shadow-api .......................... SUCCESS [  1.854 s]
[INFO] shardingsphere-shadow-distsql-handler .............. SUCCESS [ 12.647 s]
[INFO] shardingsphere-shadow-core ......................... SUCCESS [ 16.527 s]
[INFO] shardingsphere-jdbc ................................ SUCCESS [  0.191 s]
[INFO] shardingsphere-jdbc-core ........................... SUCCESS [ 55.996 s]
[INFO] shardingsphere-jdbc-spring ......................... SUCCESS [  0.190 s]
[INFO] shardingsphere-jdbc-spring-infra ................... SUCCESS [  0.200 s]
[INFO] shardingsphere-jdbc-spring-boot-starter-infra ...... SUCCESS [ 32.183 s]
[INFO] shardingsphere-jdbc-spring-namespace-infra ......... SUCCESS [ 15.967 s]
[INFO] shardingsphere-sharding-spring ..................... SUCCESS [  0.184 s]
[INFO] shardingsphere-sharding-spring-boot-starter ........ SUCCESS [ 19.790 s]
[INFO] shardingsphere-shadow-spring ....................... SUCCESS [  0.209 s]
[INFO] shardingsphere-shadow-spring-boot-starter .......... SUCCESS [ 21.554 s]
[INFO] shardingsphere-encrypt-spring ...................... SUCCESS [  0.159 s]
[INFO] shardingsphere-encrypt-spring-boot-starter ......... SUCCESS [ 16.797 s]
[INFO] shardingsphere-readwrite-splitting-spring .......... SUCCESS [  0.187 s]
[INFO] shardingsphere-readwrite-splitting-spring-boot-starter SUCCESS [ 17.259 s]
[INFO] shardingsphere-jdbc-transaction-spring ............. SUCCESS [  7.907 s]
[INFO] shardingsphere-jdbc-core-spring .................... SUCCESS [  1.163 s]
[INFO] shardingsphere-jdbc-core-spring-boot-starter ....... SUCCESS [ 34.466 s]
[INFO] shardingsphere-sharding-spring-namespace ........... SUCCESS [ 11.177 s]
[INFO] shardingsphere-readwrite-splitting-spring-namespace  SUCCESS [ 10.840 s]
[INFO] shardingsphere-encrypt-spring-namespace ............ SUCCESS [ 10.335 s]
[INFO] shardingsphere-shadow-spring-namespace ............. SUCCESS [ 10.513 s]
[INFO] shardingsphere-jdbc-core-spring-namespace .......... SUCCESS [ 38.299 s]
[INFO] shardingsphere-scaling-core ........................ SUCCESS [ 42.941 s]
[INFO] shardingsphere-scaling-distsql-handler ............. SUCCESS [ 12.323 s]
[INFO] shardingsphere-scaling-dialect ..................... SUCCESS [  0.199 s]
[INFO] shardingsphere-scaling-mysql ....................... SUCCESS [ 17.354 s]
[INFO] shardingsphere-scaling-postgresql .................. SUCCESS [ 15.642 s]
[INFO] shardingsphere-proxy ............................... SUCCESS [  0.167 s]
[INFO] shardingsphere-proxy-backend ....................... FAILURE [ 25.504 s]
[INFO] shardingsphere-proxy-frontend ...................... SKIPPED
[INFO] shardingsphere-proxy-frontend-spi .................. SKIPPED
[INFO] shardingsphere-proxy-frontend-core ................. SKIPPED
[INFO] shardingsphere-proxy-frontend-mysql ................ SKIPPED
[INFO] shardingsphere-proxy-frontend-postgresql ........... SKIPPED
[INFO] shardingsphere-proxy-frontend-opengauss ............ SKIPPED
[INFO] shardingsphere-scaling-opengauss ................... SKIPPED
[INFO] shardingsphere-proxy-bootstrap ..................... SKIPPED
[INFO] shardingsphere-db-discovery-spring ................. SKIPPED
[INFO] shardingsphere-db-discovery-spring-namespace ....... SKIPPED
[INFO] shardingsphere-db-discovery-spring-boot-starter .... SKIPPED
[INFO] shardingsphere-agent ............................... SKIPPED
[INFO] shardingsphere-agent-api ........................... SKIPPED
[INFO] shardingsphere-agent-core .......................... SKIPPED
[INFO] shardingsphere-agent-bootstrap ..................... SKIPPED
[INFO] shardingsphere-agent-plugins ....................... SKIPPED
[INFO] shardingsphere-agent-plugin-metrics ................ SKIPPED
[INFO] shardingsphere-agent-metrics-api ................... SKIPPED
[INFO] shardingsphere-agent-metrics-prometheus ............ SKIPPED
[INFO] shardingsphere-agent-plugin-tracing ................ SKIPPED
[INFO] shardingsphere-agent-tracing-test .................. SKIPPED
[INFO] shardingsphere-agent-tracing-jaeger ................ SKIPPED
[INFO] shardingsphere-agent-tracing-zipkin ................ SKIPPED
[INFO] shardingsphere-agent-tracing-opentracing ........... SKIPPED
[INFO] shardingsphere-agent-tracing-opentelemetry ......... SKIPPED
[INFO] shardingsphere-agent-plugin-logging ................ SKIPPED
[INFO] shardingsphere-agent-logging-base .................. SKIPPED
[INFO] shardingsphere-agent-distribution .................. SKIPPED
[INFO] shardingsphere-integration-test .................... SKIPPED
[INFO] shardingsphere-integration-test-fixture ............ SKIPPED
[INFO] shardingsphere-integration-test-suite .............. SKIPPED
[INFO] shardingsphere-integration-driver-test ............. SKIPPED
[INFO] shardingsphere-integration-agent-test .............. SKIPPED
[INFO] shardingsphere-integration-agent-test-plugins ...... SKIPPED
[INFO] shardingsphere-integration-agent-test-common ....... SKIPPED
[INFO] shardingsphere-integration-agent-test-metrics ...... SKIPPED
[INFO] shardingsphere-integration-agent-test-zipkin ....... SKIPPED
[INFO] shardingsphere-integration-agent-test-jaeger ....... SKIPPED
[INFO] shardingsphere-integration-agent-test-opentelemetry  SKIPPED
[INFO] shardingsphere-integration-scaling-test ............ SKIPPED
[INFO] shardingsphere-integration-scaling-test-mysql ...... SKIPPED
[INFO] shardingsphere-rewrite-test ........................ SKIPPED
[INFO] shardingsphere-optimize-test ....................... SKIPPED
[INFO] shardingsphere-distribution ........................ SKIPPED
[INFO] shardingsphere-src-distribution .................... SKIPPED
[INFO] shardingsphere-jdbc-distribution ................... SKIPPED
[INFO] shardingsphere-proxy-distribution .................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  26:53 min
[INFO] Finished at: 2021-12-06T14:06:45Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project shardingsphere-proxy-backend: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/travis/build/apache/shardingsphere/shardingsphere-proxy/shardingsphere-proxy-backend/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :shardingsphere-proxy-backend
/home/travis/.travis/functions: line 607:  8736 Terminated              travis_jigger "${!}" "${timeout}" "${cmd[@]}"
The command "travis_wait 120 ./mvnw -B --no-transfer-progress clean install cobertura:cobertura -Dmaven.javadoc.skip=true -Ddatabases=MySQL,PostgreSQL" exited with 1.
```

Tests also failed on my Ubuntu 20.04.

```
java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSetAutoCommitToOff(TextProtocolBackendHandlerFactoryTest.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithShowDatabase(TextProtocolBackendHandlerFactoryTest.java:173)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.Exception: Unexpected exception, expected<org.apache.shardingsphere.sql.parser.exception.SQLParsingException> but was<java.lang.NullPointerException>

	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithErrorRDL(TextProtocolBackendHandlerFactoryTest.java:227)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	... 22 more


java.lang.Exception: Unexpected exception, expected<org.apache.shardingsphere.sql.parser.exception.SQLParsingException> but was<java.lang.NullPointerException>

	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithErrorSQL(TextProtocolBackendHandlerFactoryTest.java:220)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	... 22 more


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithScopeSetAutoCommitToOnForInTransaction(TextProtocolBackendHandlerFactoryTest.java:159)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSet(TextProtocolBackendHandlerFactoryTest.java:180)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithUse(TextProtocolBackendHandlerFactoryTest.java:166)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithStartTransaction(TextProtocolBackendHandlerFactoryTest.java:131)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithCommonDistSQL(TextProtocolBackendHandlerFactoryTest.java:108)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithSetAutoCommitToOn(TextProtocolBackendHandlerFactoryTest.java:152)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithShow(TextProtocolBackendHandlerFactoryTest.java:187)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithBegin(TextProtocolBackendHandlerFactoryTest.java:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithQuery(TextProtocolBackendHandlerFactoryTest.java:203)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


java.lang.NullPointerException
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserExecutor.<init>(SQLStatementParserExecutor.java:37)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngine.<init>(SQLStatementParserEngine.java:35)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.lambda$getSQLStatementParserEngine$0(SQLStatementParserEngineFactory.java:43)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.shardingsphere.infra.parser.sql.SQLStatementParserEngineFactory.getSQLStatementParserEngine(SQLStatementParserEngineFactory.java:43)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.<init>(ShardingSphereSQLParserEngine.java:40)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:87)
	at org.apache.shardingsphere.proxy.backend.text.TextProtocolBackendHandlerFactoryTest.assertNewInstanceWithScopeSetAutoCommitToOff(TextProtocolBackendHandlerFactoryTest.java:145)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)


Process finished with exit code 255
```

--- a/shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/TextProtocolBackendHandlerFactoryTest.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/TextProtocolBackendHandlerFactoryTest.java
@@ -24,6 +24,8 @@ import org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData;
 import org.apache.shardingsphere.infra.metadata.rule.ShardingSphereRuleMetaData;
 import org.apache.shardingsphere.mode.manager.ContextManager;
 import org.apache.shardingsphere.mode.metadata.MetaDataContexts;
+import org.apache.shardingsphere.parser.config.SQLParserRuleConfiguration;
+import org.apache.shardingsphere.parser.rule.SQLParserRule;
 import org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.JDBCConnectionSession;
 import org.apache.shardingsphere.proxy.backend.context.ProxyContext;
 import org.apache.shardingsphere.proxy.backend.text.admin.DatabaseAdminQueryBackendHandler;
@@ -37,6 +39,7 @@ import org.apache.shardingsphere.proxy.backend.text.distsql.ral.common.ShowDistS
 import org.apache.shardingsphere.proxy.backend.text.skip.SkipBackendHandler;
 import org.apache.shardingsphere.proxy.backend.text.transaction.TransactionAutoCommitHandler;
 import org.apache.shardingsphere.proxy.backend.text.transaction.TransactionBackendHandler;
+import org.apache.shardingsphere.sql.parser.api.CacheOption;
 import org.apache.shardingsphere.sql.parser.exception.SQLParsingException;
 import org.apache.shardingsphere.transaction.ShardingSphereTransactionManagerEngine;
 import org.apache.shardingsphere.transaction.context.TransactionContexts;
@@ -50,6 +53,7 @@ import org.mockito.junit.MockitoJUnitRunner;
 
 import java.sql.SQLException;
 import java.util.Collections;
+import java.util.Optional;
 import java.util.Properties;
 
 import static org.hamcrest.CoreMatchers.instanceOf;
@@ -80,6 +84,8 @@ public final class TextProtocolBackendHandlerFactoryTest {
         when(metaDataContexts.getProps()).thenReturn(new ConfigurationProperties(new Properties()));
         TransactionContexts transactionContexts = mockTransactionContexts();
         when(contextManager.getTransactionContexts()).thenReturn(transactionContexts);
+        CacheOption cacheOption = new CacheOption(1024, 1024, 1024);
+        when(metaDataContexts.getGlobalRuleMetaData().findSingleRule(SQLParserRule.class)).thenReturn(Optional.of(new SQLParserRule(new SQLParserRuleConfiguration(true, cacheOption, cacheOption))));
         ProxyContext.getInstance().init(contextManager);
     }
     

